test/Serialization/target-incompatible.swift

```diff
@@ -8,6 +8,13 @@
 // RUN: %{python} -u %S/Inputs/binary_sub.py x86_64-unknown-darwin14 x86_64-unknown-solaris8 < %t/solaris-template.swiftmodule > %t/solaris.swiftmodule
 // RUN: not %target-swift-frontend -I %t -typecheck -parse-stdlib %s -DSOLARIS 2>&1 | %FileCheck -check-prefix=CHECK-SOLARIS %s
 
+// Check that we still get the diagnostic but the module is output anyway when
+// allowing errors
+// RUN: %target-swift-frontend -I %t -parse-stdlib -experimental-allow-module-with-compiler-errors -emit-module -module-name incompatmips -o %t %s -DMIPS 2>&1 | %FileCheck -check-prefix=CHECK-MIPS %s
+// RUN: ls %t/incompatmips.swiftmodule
+// RUN: %target-swift-frontend -I %t -parse-stdlib -experimental-allow-module-with-compiler-errors -emit-module -module-name incompatsol -o %t %s -DSOLARIS 2>&1 | %FileCheck -check-prefix=CHECK-SOLARIS %s
+// RUN: ls %t/incompatsol.swiftmodule
```

Comment:
Are these `ls` to check that the files were created?

Reply:
Yep, I've used llvm-bcanalyzer elsewhere with a check for `UnknownCode`. Happy to do that here as well, just figured it was overkill.